src/internal/connector/data_collections.go

```diff
@@ -121,6 +121,7 @@ func (gc *GraphConnector) ProduceBackupCollections(
 	case selectors.ServiceSharePoint:
 		colls, ssmb, err = sharepoint.DataCollections(
 			ctx,
+			gc.Discovery,
```

Comment:
Just a general comment - We are passing 4 `gc` fields in here. If we always use these fields together, it might be better to containerize them together. It will reduce the # of args, plus improve readability

Reply:
agreed.  that's a refactoring pass which is out of scope for these changes, but is something we could trickle through alongside other work.